shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/ParquetCachedBatchSerializer.scala

```diff
@@ -244,14 +303,123 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
    * @return RDD of the rows that were stored in the cached batches.
    */
   override def convertCachedBatchToInternalRow(input: RDD[CachedBatch],
-                                cacheAttributes: Seq[Attribute],
-                                selectedAttributes: Seq[Attribute],
-                                conf: SQLConf): RDD[InternalRow] = {
-    val cb = convertCachedBatchToColumnarBatch(input, cacheAttributes, selectedAttributes, conf)
-    val rowRdd = cb.mapPartitions(iter => {
-      new ColumnarToRowIterator(iter)
-    })
-    rowRdd
+     cacheAttributes: Seq[Attribute],
+     selectedAttributes: Seq[Attribute],
+     conf: SQLConf): RDD[InternalRow] = {
+    val rapidsConf = new RapidsConf(conf)
+    if (rapidsConf.isSqlEnabled) {
```

Comment:
So to be clear I don't think there is a ever a situation where we want the GPU enabled version of this.

I thought that `GpuInMemoryTableScan` would call a GPU only version of reading, which would by design be columnar, and that `InMemoryTableScan` would call the regular APIs. But then it does not matter if the plugin is enabled, if we cannot support the data types/etc we should not use the GPU, which is the point of having a separate API for the GPU version.  It also lets us do things like grab the GPU Semaphore to be sure that we have it when processing data on the GPU.

Reply:
Yes, you are right. I forgot about the implicit checks on the execs. Let me make the change